### PR TITLE
232 revises resize/zoom

### LINKED ISSUE #290: Optional dependency cupy is not tested normally
**Describe the bug**
The cupy library is used to do zooming on GPU as an optional requirement, if not installed scipy is used. The requirements.txt doesn't mention it so the test setup doesn't install it, leading to a number of test cases to do nothing rather than test the functionality. 

One bug that is present is the `test_gpu_zoom` test in `tests/test_rand_zoomd.py` where line 63 should be:

```python
self.assertTrue(np.allclose(expected, zoomed[key]))
````

**To Reproduce**
1. Setup the environment with Python 3.7 and the libraries in requirements.txt
2. Install `cupy`
3. Run `./runtests.sh`

**Expected behavior**
This is the question: 
* Should the CI tests be setup to install cupy and run those tests, or
* Should these test be left as optional?

**Environment (please complete the following information):**
 - Ubuntu 16.04
 - Python 3.7
 - MONAI 0.1.0
 - CUDA/cuDNN 10.1

### PR DESCRIPTION
**Pull request by wyli**
subtask of #232; fixes #290 

Revises zoom/zoomd, resize/resized, randzoom/randzoomd, png_saver to use torch interpolate instead of skimage.transform 

### Status
**ready**

### Types of changes

- [x] Breaking change (fix or new feature that would cause existing functionality to change)
- [x] New tests added to cover the changes
- [x] Docstrings/Documentation updated

### CODE REVIEW

**Comment by Nic-Ma** [comment]
> @wyli : FYI: The master branch also fails at git hash `7444466cd3d1782444dc98d13d0e8c3ff7add191`
> 
> ```
> python -m tests.test_integration_workflows.IntegrationWorkflows
> ```
> 
> ```
> FAIL: test_training (__main__.IntegrationWorkflows)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/home/johnsonhj/src/monai/tests/test_integration_workflows.py", line 276, in test_training
>     np.testing.assert_allclose(infer_metric, 0.9224808603525162, rtol=1e-3)
>   File "/home/johnsonhj/.venv/monai37/lib/python3.7/site-packages/numpy/testing/_private/utils.py", line 1533, in assert_allclose
>     verbose=verbose, header=header, equal_nan=equal_nan)
>   File "/home/johnsonhj/.venv/monai37/lib/python3.7/site-packages/numpy/testing/_private/utils.py", line 846, in assert_array_compare
>     raise AssertionError(msg)
> AssertionError: 
> Not equal to tolerance rtol=0.001, atol=0
> 
> Mismatched elements: 1 / 1 (100%)
> Max absolute difference: 0.01481684
> Max relative difference: 0.01606195
>  x: array(0.937298)
>  y: array(0.922481)
> 
> ----------------------------------------------------------------------
> Ran 1 test in 45.166s
> ```

Hi @hjmjohnson ,

Thanks for your help here.
I am fixing the integration test error in PR #534

**Comment by Nic-Ma** [comment]
BTW, I ran the `test_integration_classification_2d` test, it passed as expected:
```
python tests/test_integration_classification_2d.py
2020-06-11 08:54:38 URL:https://uc1dda3a4e4c1763afdb3fb2e06c.dl.dropboxusercontent.com/cd/0/inline2/A5eCNnYGNh3H2m7g9NCUbctXqqPerwD9LehHUTzpnX-25Ry7_RkOo-2ZjEXK2S4Y0wM4S252LN2cOSGzHTYAIpmvM9y_nN--0X4I6uDwexCX3lPZWUBHAUgS45nRepwZd9ljHNfsnWhD2fm8LairrKIOnr7Se-w56HR8uN4AySNEEja0WMn90P7FSUy-SqCksUxG5hXzW7wg2WezYpVKDaIqfWYJ_Tv0OGoDt5jQkyKJamDwTBEh2IixWGdYhz-ITB42MrTF22Ms2mePcx6MiVtjIvWsZhebk3pO0Ch1K_JZPwpyXwhNXB-N1I3rpELdFPRvcVxtMGeRv2wnX-XL4Cb7l9fdF-NZZ_yW9V_FZ5oytw/file [61834679/61834679] -> "/tmp/tmpx94_q0aq/MedNIST.tar.gz" [1]
MONAI version: 0+untagged.263.g5c85522
Python version: 3.6.9 |Anaconda, Inc.| (default, Jul 30 2019, 19:07:31)  [GCC 7.3.0]
Numpy version: 1.17.4
Pytorch version: 1.5.0
Ignite version: 0.3.0
----------
Epoch 1/4
epoch 1 average loss:0.8184
saved new best metric model
current epoch 1 current AUC: 0.9979 current accuracy: 0.9641 best AUC: 0.9979 at epoch 1
----------
Epoch 2/4
epoch 2 average loss:0.1730
saved new best metric model
current epoch 2 current AUC: 0.9996 current accuracy: 0.9854 best AUC: 0.9996 at epoch 2
----------
Epoch 3/4
epoch 3 average loss:0.0779
saved new best metric model
current epoch 3 current AUC: 0.9998 current accuracy: 0.9909 best AUC: 0.9998 at epoch 3
----------
Epoch 4/4
epoch 4 average loss:0.0480
saved new best metric model
current epoch 4 current AUC: 0.9999 current accuracy: 0.9942 best AUC: 0.9999 at epoch 4
train completed, best_metric: 0.9999  at epoch: 4
best metric 0.9999425584560568
MONAI version: 0+untagged.263.g5c85522
Python version: 3.6.9 |Anaconda, Inc.| (default, Jul 30 2019, 19:07:31)  [GCC 7.3.0]
Numpy version: 1.17.4
Pytorch version: 1.5.0
Ignite version: 0.3.0
----------
Epoch 1/4
epoch 1 average loss:0.8184
saved new best metric model
current epoch 1 current AUC: 0.9979 current accuracy: 0.9641 best AUC: 0.9979 at epoch 1
----------
Epoch 2/4
epoch 2 average loss:0.1730
saved new best metric model
current epoch 2 current AUC: 0.9996 current accuracy: 0.9854 best AUC: 0.9996 at epoch 2
----------
Epoch 3/4
epoch 3 average loss:0.0779
saved new best metric model
current epoch 3 current AUC: 0.9998 current accuracy: 0.9909 best AUC: 0.9998 at epoch 3
----------
Epoch 4/4
epoch 4 average loss:0.0480
saved new best metric model
current epoch 4 current AUC: 0.9999 current accuracy: 0.9942 best AUC: 0.9999 at epoch 4
train completed, best_metric: 0.9999  at epoch: 4
best metric 0.9999425584560568
.
----------------------------------------------------------------------
Ran 1 test in 262.239s

OK
```

**Comment by wyli** [comment]
> BTW, I ran the `test_integration_classification_2d` test, it passed as expected:
> 
> ```
> python tests/test_integration_classification_2d.py
> 2020-06-11 08:54:38 URL:https://uc1dda3a4e4c1763afdb3fb2e06c.dl.dropboxusercontent.com/cd/0/inline2/A5eCNnYGNh3H2m7g9NCUbctXqqPerwD9LehHUTzpnX-25Ry7_RkOo-2ZjEXK2S4Y0wM4S252LN2cOSGzHTYAIpmvM9y_nN--0X4I6uDwexCX3lPZWUBHAUgS45nRepwZd9ljHNfsnWhD2fm8LairrKIOnr7Se-w56HR8uN4AySNEEja0WMn90P7FSUy-SqCksUxG5hXzW7wg2WezYpVKDaIqfWYJ_Tv0OGoDt5jQkyKJamDwTBEh2IixWGdYhz-ITB42MrTF22Ms2mePcx6MiVtjIvWsZhebk3pO0Ch1K_JZPwpyXwhNXB-N1I3rpELdFPRvcVxtMGeRv2wnX-XL4Cb7l9fdF-NZZ_yW9V_FZ5oytw/file [61834679/61834679] -> "/tmp/tmpx94_q0aq/MedNIST.tar.gz" [1]
> MONAI version: 0+untagged.263.g5c85522
> Python version: 3.6.9 |Anaconda, Inc.| (default, Jul 30 2019, 19:07:31)  [GCC 7.3.0]
> Numpy version: 1.17.4
> Pytorch version: 1.5.0
> Ignite version: 0.3.0
> ----------
> Epoch 1/4
> epoch 1 average loss:0.8184
> saved new best metric model
> current epoch 1 current AUC: 0.9979 current accuracy: 0.9641 best AUC: 0.9979 at epoch 1
> ----------
> Epoch 2/4
> epoch 2 average loss:0.1730
> saved new best metric model
> current epoch 2 current AUC: 0.9996 current accuracy: 0.9854 best AUC: 0.9996 at epoch 2
> ----------
> Epoch 3/4
> epoch 3 average loss:0.0779
> saved new best metric model
> current epoch 3 current AUC: 0.9998 current accuracy: 0.9909 best AUC: 0.9998 at epoch 3
> ----------
> Epoch 4/4
> epoch 4 average loss:0.0480
> saved new best metric model
> current epoch 4 current AUC: 0.9999 current accuracy: 0.9942 best AUC: 0.9999 at epoch 4
> train completed, best_metric: 0.9999  at epoch: 4
> best metric 0.9999425584560568
> MONAI version: 0+untagged.263.g5c85522
> Python version: 3.6.9 |Anaconda, Inc.| (default, Jul 30 2019, 19:07:31)  [GCC 7.3.0]
> Numpy version: 1.17.4
> Pytorch version: 1.5.0
> Ignite version: 0.3.0
> ----------
> Epoch 1/4
> epoch 1 average loss:0.8184
> saved new best metric model
> current epoch 1 current AUC: 0.9979 current accuracy: 0.9641 best AUC: 0.9979 at epoch 1
> ----------
> Epoch 2/4
> epoch 2 average loss:0.1730
> saved new best metric model
> current epoch 2 current AUC: 0.9996 current accuracy: 0.9854 best AUC: 0.9996 at epoch 2
> ----------
> Epoch 3/4
> epoch 3 average loss:0.0779
> saved new best metric model
> current epoch 3 current AUC: 0.9998 current accuracy: 0.9909 best AUC: 0.9998 at epoch 3
> ----------
> Epoch 4/4
> epoch 4 average loss:0.0480
> saved new best metric model
> current epoch 4 current AUC: 0.9999 current accuracy: 0.9942 best AUC: 0.9999 at epoch 4
> train completed, best_metric: 0.9999  at epoch: 4
> best metric 0.9999425584560568
> .
> ----------------------------------------------------------------------
> Ran 1 test in 262.239s
> 
> OK
> ```

thank you!

**Comment by wyli** [comment]
> Looks good to me.
> And I am wondering why you don't use the `AffineTransform` to do resize/zoom? I think it can also update the affine matrix.

yes, it's the same logic but nn.functional.interpolate supports more interpolation mode(, and probably faster)

File: monai/handlers/segmentation_saver.py [hunk 1]
```diff
@@ -43,19 +43,16 @@ def __init__(
...
             resample (bool): whether to resample before saving the data array.
-            interp_order (int or str):
-                The interpolation mode. This option is used when `resample = True`.
-                When saving NIfTI files, the options are "nearest", "bilinear". Defaults to "bilinear".
-                See also: https://pytorch.org/docs/stable/nn.functional.html#grid-sample
-                When saving PNG files, this sets the order of the spline interpolation.
-                The order has to be in the range 0 - 5.
-                https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.affine_transform.html
+            interp_order (str):
+                The interpolation mode. Defaults to "nearest". This option is used when `resample = True`.
```
[the remark this answers is not in the document]
Comment (wyli) [reply monai/handlers/segmentation_saver.py:47]: thanks, I updated the code

File: monai/data/png_writer.py [hunk 2]
```diff
@@ -50,17 +40,21 @@ def write_png(
...
         if len(data.shape) == 3:
-            output_shape = tuple(output_shape) + (data.shape[2],)
-
-        data = transform.resize(data, output_shape, order=interp_order, mode=mode, cval=cval, preserve_range=True)
+            data = np.moveaxis(data, -1, 0)  # to channel first
+            data = xform(data)
+            data = np.moveaxis(data, 0, -1)
+        else:
+            data = np.expand_dims(data, 0)  # make a channel
+            data = xform(data)[0]  # first batch first channel
```
[the remark this answers is not in the document]
Comment (wyli) [reply monai/data/png_writer.py]: thanks, I updated the code

**Pull request merged**